app/services/between_categories_transactions/creator.rb

```diff
@@ -0,0 +1,54 @@
+class BetweenCategoriesTransactions::Creator
+  def initialize(amount:, user_id:, category_from_id:, category_to_id:)
+    @amount = amount
+    @user_id = user_id
+    @category_from_id = category_from_id
+    @category_to_id = category_to_id
+  end
+
+  def create
+    return unless valid?
+    ActiveRecord::Base.transaction do
+      create_between_category_transaction
+      update_categories_balance
+    end
+  end
+
+  private
+
+  def create_between_category_transaction
+    transaction = BetweenCategoriesTransaction.create(category_from_id: @category_from_id,
+                                                      category_to_id: @category_to_id)
+    transaction.transactions.create(amount: @amount, user_id: @user_id,
+                                    date: Date.current)
+  end
```

Comment:
@marderer I've noticed, that this logic can be organized like in several places, for example:
1) `BetweenCategoriesTransactions::Creator`
```ruby
def create_between_category_transaction
  transaction = BetweenCategoriesTransaction.create(
    category_from_id: @category_from_id,
    category_to_id: @category_to_id
  )
  transaction.transactions.create(
    amount: @amount, 
    user_id: @user_id,
    date: Date.current
  )
end
```
2) `BalanceTransactions::Creator`
```ruby
  def create_balance_transaction
    @transaction = BalanceTransaction.create(comment: @comment)
  end

  def add_transaction_to_user
    @transaction.transactions.create(amount: @amount, date: @date.to_date,
                                     user_id: @user_id)
  end
```

I didn't do two methods for creation - only one. I guess it's because you didn't want to overload service more.

What do you think about splitting it then:
```ruby
class BetweenCategoriesTransactions::Creator
  def initialize(amount:, user_id:, category_from_id:, category_to_id:)
    @amount = amount
    @user_id = user_id
    @category_from_id = category_from_id
    @category_to_id = category_to_id
  end

  def create
    return unless valid?
    ActiveRecord::Base.transaction do
      create_between_category_transaction
      create_transaction
      transfer_money_between_categories
    end
  end

  private

  def create_between_category_transaction
    @transaction = BetweenCategoriesTransaction.create(
      category_from_id: @category_from_id,
      category_to_id: @category_to_id
    )
  end

  def create_transaction
    @transaction.transactions.create(
      amount: @amount, 
      user_id: @user_id,
      date: Date.current
    )
  end

  def transfer_money_between_categories
    Categories::BalanceTransferer.new(
      category_from: category_from,
      category_to: category_to,
      amount: @amount
    ).transfer
  end

  def valid?
    category_from.amount >= @amount
  end

  def category_from
    @_category_from ||= user_categories.find(@category_from_id)
  end

  def category_to
    @_category_to ||= user_categories.find(@category_to_id)
  end

  def user
    @_user ||= User.find(@user_id)
  end

  delegate :categories, to: :user, prefix: true
end

class Categories::BalanceTransferer
  def initialize(...)
    ...
  end

  def transfer
    category_from.update(amount: category_from_amount)
    category_to.update(amount: category_to_amount)
  end

  private

  def category_from_amount
    category_from.amount - @amount
  end

  def category_to_amount
    category_to.amount + @amount
  end
end
```